packages/logging/service.go

```diff
@@ -89,6 +128,71 @@ func (s *loggingService) GetForDogu(request *pb.DoguLogMessageRequest, server pb
 	return writeLogLinesToStream(s.logProvider, doguName, linesCount, server)
 }
 
+func (s *loggingService) SetLogLevel(ctx context.Context, req *pb.LogLevelRequest) (*emptypb.Empty, error) {
+	doguName := req.DoguName
+
+	if strings.TrimSpace(doguName) == "" {
+		return nil, createInternalErr(errors.New(responseMessageMissingDoguname), codes.InvalidArgument)
+	}
+
+	lLevel, err := mapLogLevel(req.GetLogLevel())
+	if err != nil {
+		return nil, createInternalErr(fmt.Errorf("unable to map log level from proto message: %w", err), codes.InvalidArgument)
+	}
+
+	restart, err := s.setLogLevel(ctx, doguName, lLevel)
+	if err != nil {
+		return nil, createInternalErr(fmt.Errorf("unable to set log level: %w", err), codes.Internal)
+	}
+
+	if !restart {
+		return &emptypb.Empty{}, nil
+	}
+
+	if lErr := s.doguRestarter.RestartDogu(ctx, doguName); lErr != nil {
+		return nil, createInternalErr(fmt.Errorf("unable to restart dogu %s after setting new log level: %w", doguName, lErr), codes.Internal)
+	}
+
+	return &emptypb.Empty{}, nil
+}
+
+func (s *loggingService) setLogLevel(_ context.Context, doguName string, l logLevel) (bool, error) {
+	const loggingKey = "logging/root"
+
+	dConfig := s.configProvider.DoguConfig(doguName)
+
+	currentLevel, err := dConfig.Get(loggingKey)
```

Comment:
This needs a tiny change to accommodate for empty (aka: default) values in `logging/root`. We probably don't want to restart a dogu when a default value (of say: error) will explicitly set (to error in this case) and the dogu restarts. We could avoid this downtime too.